File: clinic-backend/src/main/java/com/clinicapp/backend/dto/notification/NotificationDTO.java
package com.clinicapp.backend.dto.notification;

import com.clinicapp.backend.model.notification.Notification;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class NotificationDTO {
    private Long id;
    private Notification.NotificationType type;
    private String title;
    private String message;
    private Instant createdAt;
    private boolean read;
    private String referenceType;
    private Long referenceId;
    private String sender; // NEW: sender's email
    private String recipient; // NEW: recipient's email
}

File: clinic-frontend1/src/app/models/notification.dto.ts
// Matches com.clinicapp.backend.dto.notification.NotificationDTO

// Ensure this matches the NotificationType enum in the backend
export type NotificationType =
  | 'APPOINTMENT_REMINDER'
  | 'NEW_MESSAGE'
  | 'PRESCRIPTION_READY'
  | 'INVOICE_GENERATED'
  | 'APPOINTMENT_CANCELLED'
  | 'APPOINTMENT_MODIFIED'
  | 'SYSTEM_NOTIFICATION';

export interface NotificationDTO {
  id: number;
  type: NotificationType;
  title: string;
  message: string;
  createdAt: string; // ISO string format
  read: boolean;
  referenceType?: string; // e.g., "APPOINTMENT", "PRESCRIPTION", "MESSAGE"
  referenceId?: number;
  sender?: string; // who sent the notification
  recipient?: string; // who should receive it
}